src/core/algorithms/dc/FastADC/util/approximate_evidence_inverter.h

```diff
@@ -0,0 +1,259 @@
+    PredicateIndexProvider* predicate_index_provider_;
+
+    struct SearchNode {
+        size_t e;
+        boost::dynamic_bitset<> addable_predicates;
+        DCCandidateTrie dc_candidates;
+        std::vector<DCCandidate> invalid_dcs;
+        int64_t target;
+
+        SearchNode(size_t e, boost::dynamic_bitset<> const& addable_predicates,
+                   DCCandidateTrie&& dc_candidates, std::vector<DCCandidate> const& invalid_dcs,
+                   int64_t target)
+            : e(e),
+              addable_predicates(addable_predicates),
+              dc_candidates(std::move(dc_candidates)),
+              invalid_dcs(invalid_dcs),
+              target(target) {}
+    };
+
+    void InverseEvidenceSet() {
+        LOG(DEBUG) << "  [AEI] Inverting evidences...";
+
+        approx_covers_ = DCCandidateTrie(n_predicates_);
+        boost::dynamic_bitset<> full_mask(n_predicates_);
+        full_mask.set();
+        std::stack<SearchNode> nodes;  // Manual stack, where evidences_[node.e] needs to be hit
+
+        DCCandidateTrie dc_candidates(n_predicates_);
+        dc_candidates.Add(DCCandidate{.cand = full_mask});
+
+        Walk(0, full_mask, std::move(dc_candidates), target_, nodes);
+
+        while (!nodes.empty()) {
+            SearchNode nd = std::move(nodes.top());
+            nodes.pop();
+            if (nd.e >= evidences_.size() || nd.addable_predicates.none()) continue;
+            Hit(nd);  // Hit evidences_[e]
+            if (nd.target > 0)
+                Walk(nd.e + 1, nd.addable_predicates, std::move(nd.dc_candidates), nd.target,
+                     nodes);
+        }
+    }
+
+    void Walk(size_t e, boost::dynamic_bitset<>& addable_predicates,
+              DCCandidateTrie&& dc_candidates, int64_t target, std::stack<SearchNode>& nodes) {
+        while (e < evidences_.size() && !dc_candidates.IsEmpty()) {
+            PredicateBitset evi = evidences_[e].evidence;
+            auto unhit_evi_dcs = dc_candidates.GetAndRemoveGeneralizations(evi);
+
+            // Hit evidences_[e] later
+            nodes.push(SearchNode(e, addable_predicates, std::move(dc_candidates), unhit_evi_dcs,
+                                  target));
+
+            // Unhit evidences_[e]
+            if (unhit_evi_dcs.empty()) return;
+
+            addable_predicates &= evi;
+
+            if (addable_predicates.none()) return;
+
+            int64_t max_can_hit = 0;
+            for (size_t i = e + 1; i < evidences_.size(); ++i)
+                if (!(IsSubset(addable_predicates, evidences_[i].evidence)))
+                    max_can_hit += evidences_[i].count;
+            if (max_can_hit < target) return;
+
+            DCCandidateTrie new_candidates(n_predicates_);
+            for (auto const& dc : unhit_evi_dcs) {
+                boost::dynamic_bitset<> unhit_cand = dc.cand & evi;
+                if (unhit_cand.any())
+                    new_candidates.Add(DCCandidate(dc.bitset, unhit_cand));
```

Comment:
```suggestion
                    new_candidates.Add(DCCandidate{dc.bitset, unhit_cand});
```

Parenthised aggregate initialization isn't currenlty supported by Apple Clang. Only braced one is allowed.